### PR TITLE
More Dangerous Mindbreaking

### PR DESCRIPTION
**Pull request by VMSolidus**
# Description

Somehow people weren't getting the memo despite the **HORRIFYING** texts associated with Mindbreaking, that doing so is an almost comically evil act akin to murder. So this PR fixes that. By making it even more obvious.

<details><summary><h1>Media</h1></summary>
<p>

image

</p>
</details>

# Changelog

:cl:
- add: Added a server option for "Scarier Mindbreaking". Mindbreaking now irreversibly converts a player character into a non-sentient NPC.
- fix: Mindbreaking now only works on Psychics.

### CODE REVIEW

**Comment by bruhmogus** [comment]
will this be on or off by default, since LSD + hypospray is able to really fuck up balance (just a tad)

**Comment by VMSolidus** [comment]
> will this be on or off by default, since LSD + hypospray is able to really fuck up balance (just a tad)

Just don't take the Latent Psychic trait. 
faridaiscute

**Comment by bruhmogus** [comment]
> > will this be on or off by default, since LSD + hypospray is able to really fuck up balance (just a tad)
> 
> Just don't take the Latent Psychic trait.

iirc you can mindbreak someone who isnt even a psionic

**Comment by VMSolidus** [comment]
> > > will this be on or off by default, since LSD + hypospray is able to really fuck up balance (just a tad)
> > 
> > 
> > Just don't take the Latent Psychic trait.
> 
> iirc you can mindbreak someone who isnt even a psionic

I'm going to fix that *right now*

**Comment by VMSolidus** [comment]
> > > will this be on or off by default, since LSD + hypospray is able to really fuck up balance (just a tad)
> > 
> > 
> > Just don't take the Latent Psychic trait.
> 
> iirc you can mindbreak someone who isnt even a psionic

Done.
image

**Comment by bruhmogus** [comment]
also how will this play with the trait that literally gives you the mindbroken component roundstart

**Comment by VMSolidus** [comment]
> also how will this play with the trait that literally gives you the mindbroken component roundstart

It doesn't. 
Chadly

**Comment by bruhmogus** [comment]
> > also how will this play with the trait that literally gives you the mindbroken component roundstart
> 
> It doesn't. 

hm. maybe it should, but at the same time it'd be hilarious to instantly get ghosted for having the trait

**Comment by gluesniffler** [comment]
> > > also how will this play with the trait that literally gives you the mindbroken component roundstart
> > 
> > 
> > It doesn't.
> 
> hm. maybe it should, but at the same time it'd be hilarious to instantly get ghosted for having the trait

It's better this way, because then you have to doubt whether or not the person you're talking to is an NPC or not IC, giving you more leeway to abuse them (killing a mindbroken person at best counts as animal abuse)

**Comment by VMSolidus** [comment]
> > > > also how will this play with the trait that literally gives you the mindbroken component roundstart
> > > 
> > > 
> > > It doesn't.
> > 
> > 
> > hm. maybe it should, but at the same time it'd be hilarious to instantly get ghosted for having the trait
> 
> It's better this way, because then you have to doubt whether or not the person you're talking to is an "NPC" or not IC, giving you more leeway to abuse them (killing a mindbroken person at best counts as animal abuse)

It would be even funnier if we rebased the NPCConversationSystem from Nyano, and/or make the mindbroken NPCs talk using ChatGPT2.

**Comment by bruhmogus** [comment]
> > > > also how will this play with the trait that literally gives you the mindbroken component roundstart
> > > 
> > > 
> > > It doesn't.
> > 
> > 
> > hm. maybe it should, but at the same time it'd be hilarious to instantly get ghosted for having the trait
> 
> It's better this way, because then you have to doubt whether or not the person you're talking to is an "NPC" or not IC, giving you more leeway to abuse them (killing a mindbroken person at best counts as animal abuse)

this confuses me, have we tested what actually happens if someone has the instant mindbreak trait? will it insta-ghost? will it let them keep playing?

**Comment by VMSolidus** [comment]
> > > > > also how will this play with the trait that literally gives you the mindbroken component roundstart
> > > > 
> > > > 
> > > > It doesn't.
> > > 
> > > 
> > > hm. maybe it should, but at the same time it'd be hilarious to instantly get ghosted for having the trait
> > 
> > 
> > It's better this way, because then you have to doubt whether or not the person you're talking to is an "NPC" or not IC, giving you more leeway to abuse them (killing a mindbroken person at best counts as animal abuse)
> 
> this confuses me, have we tested what actually happens if someone has the instant mindbreak trait? will it insta-ghost? will it let them keep playing?

It doesn't lmao. It doesn't even call this function.

**Comment by bruhmogus** [comment]
> > > > > > also how will this play with the trait that literally gives you the mindbroken component roundstart
> > > > > 
> > > > > 
> > > > > It doesn't.
> > > > 
> > > > 
> > > > hm. maybe it should, but at the same time it'd be hilarious to instantly get ghosted for having the trait
> > > 
> > > 
> > > It's better this way, because then you have to doubt whether or not the person you're talking to is an "NPC" or not IC, giving you more leeway to abuse them (killing a mindbroken person at best counts as animal abuse)
> > 
> > 
> > this confuses me, have we tested what actually happens if someone has the instant mindbreak trait? will it insta-ghost? will it let them keep playing?
> 
> It doesn't lmao. It doesn't even call this function.

oh! thats perfect! as someone with Zero Maintainer Authority™ i'd say this works really well

**Comment by Erisfiregamer1** [comment]
> > > > > also how will this play with the trait that literally gives you the mindbroken component roundstart
> > > > 
> > > > 
> > > > It doesn't.
> > > 
> > > 
> > > hm. maybe it should, but at the same time it'd be hilarious to instantly get ghosted for having the trait
> > 
> > 
> > It's better this way, because then you have to doubt whether or not the person you're talking to is an "NPC" or not IC, giving you more leeway to abuse them (killing a mindbroken person at best counts as animal abuse)
> 
> It would be even funnier if we rebased the NPCConversationSystem from Nyano, and/or make the mindbroken NPCs talk using ChatGPT2.

Just use the free Groq API and something like Llama3 on the server, I could probably help you write the code for it. It'd be funny!!

File: Content.Shared/CCVar/CCVars.cs [hunk 1]
```diff
@@ -2656,13 +2656,24 @@ public static readonly CVarDef<float>
...
+
+        #region Psionics
+
+        /// <summary>
+        ///     When mindbroken, permanently eject the player from their own body, and turn their character into an NPC.
+        ///     Congratulations, now they *actually* aren't a person anymore.
+        ///     For people who complained that it wasn't obvious enough from the text that Mindbreaking is a form of Murder.
+        /// </summary>
+        public static readonly CVarDef<bool> ScarierMindbreaking =
+            CVarDef.Create("psionics.scarier_mindbreaking", true, CVar.SERVERONLY);
```
Comment (FoxxoTrystan) [inline Content.Shared/CCVar/CCVars.cs:2676]:
```suggestion
            CVarDef.Create("psionics.scarier_mindbreaking", false, CVar.SERVERONLY);
```

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]
Make it false a lot of servers will not enjoy this change if they update and they dont know about it, this should be toggled true on server choices.

**Review by sleepyyapril — APPROVED** [review]

**Comment by bruhmogus** [comment]
mindbreaking's punishment should be changed if this is merged, as it basically kills them (and might be prevention of revival?)

**Comment by VMSolidus** [comment]
> 
> 
> mindbreaking's punishment should be changed if this is merged, as it basically kills them (and might be prevention of revival?)

I completely forgot this DeltaV feature even exists lmao. Yea mindbreaking people on EE is not like mindbreaking people on DeltaV. The design intention behind it is that mindbreaking someone is essentially a "Metaphysical" form of murder, and thus  should be reserved only as a capital punishment, or as Self-Defense in the case of being attacked by someone with psionic powers.

**Comment by bruhmogus** [comment]
> > 
> > mindbreaking's punishment should be changed if this is merged, as it basically kills them (and might be prevention of revival?)
> 
> I completely forgot this DeltaV feature even exists lmao. Yea mindbreaking people on EE is not like mindbreaking people on DeltaV. The design intention behind it is that mindbreaking someone is essentially a "Metaphysical" form of murder, and thus should be reserved only as a capital punishment, or as Self-Defense in the case of being attacked by someone with psionic powers.

yeah what im saying is just "this is literal murder, and should be treated as such"

**Comment by RustedBR** [comment]
Isn't this, for the player, the same thing as getting sacrificed as a psion?

**Review by DEATHB4DEFEAT — APPROVED** [review]

**Pull request merged**